faculty/datasets/__init__.py

```diff
@@ -481,30 +421,24 @@ def rmdir(project_path, project_id=None):
         The project to get files from. You need to have access to this project
         for it to work. Defaults to the project set by FACULTY_PROJECT_ID in
         your environment.
+    object_client : faculty.clients.object.ObjectClient, optional
+        Advanced - can be used to benefit from caching in chain interactions
+        with datasets.
     """
 
-    s3_client = _s3_client(project_id)
-
-    if not _isdir(project_path, project_id, s3_client):
-        raise DatasetsError("not a directory")
-
     contents = ls(
-        project_path, project_id, show_hidden=True, s3_client=s3_client
+        prefix=project_path,
```

Comment:
What if we list with project_path `/path/to/dir` or `/path/to/dir/` and get back the directory as an object?

Reply:
Also, I believe this method will allow you to delete files, which would be wrong.